### PR TITLE
Status and logs fix

### PR DESCRIPTION
**Pull request by swanijam**

### CODE REVIEW

File: site.py [hunk 1]
```diff
@@ -43,7 +43,7 @@ def serve_static(path):
 
 
 # Retrieve the files uploaded by the user, and if they are valid,
-# store them in a subfolder in the containers directory.
+# store them in a subfolder in the
```
Comment (bruth) [inline site.py]: Finish sentence.

File: templates/status.html [hunk 2]
```diff
@@ -31,10 +20,11 @@ <h2 id="cid">Container: {{ cid }}</h2>
                  xmlhttp.onreadystatechange = function() {
                    if (xmlhttp.readyState === 4 && xmlhttp.status === 200) {
                      var data = JSON.parse(xmlhttp.responseText);
-                     document.getElementById('status').innerHTML = data.status;
-                     document.getElementById('stdout').innerHTML = data.out;
-                     document.getElementById('stderr').innerHTML = data.err;
-                     document.getElementById('cid').innerHTML = 'Container: ' + data.cid;
+                     document.getElementById('status').innerHTML = 'Status: '
+                         + data.status;
```
Comment (bruth) [inline templates/status.html]: Move this content up to the previous line.

File: templates/status.html [hunk 3]
```diff
@@ -31,10 +20,11 @@ <h2 id="cid">Container: {{ cid }}</h2>
...
+                     if (data.status === 'Running') {
+                       window.location='/containers/' + '{{ uuid }}';
```
Comment (bruth) [inline templates/status.html]: Add spaces around the equals sign.

File: site.py [hunk 4]
```diff
@@ -315,11 +320,13 @@ def check_container(cid):
 # Updated using the function above.
 @app.route('/containers/<uuid>/status', methods=['GET'])
 def container_status(uuid):
-    cid = container_id(uuid)
-    url = '/check_container/' + cid[:16]
+    url = launch_container(uuid)
```
Comment (bruth) [inline site.py:328]: Although not obvious, this is a clever solution that bypasses the need to use a separate thread. For this to work reliably, the folder named after the [`cid`](https://github.com/swanijam/HaaS/blob/status-and-logs-fix/site.py#L63) should be checked for before this line to ensure the files exist for the generated UUID. Something like:

``` python
folder = os.path.join(CONTAINER_DIR, uuid)
if not os.path.exists(folder):
    return '', 404
```

In which case the we page polling this status should display a friendly _No container exists_ message.

Comment (swanijam) [inline site.py:328]: The check_container() method already returns a status 'Missing' if the container is not found.

Comment (bruth) [inline site.py:328]: Right, but calling [`launch_container`](https://github.com/swanijam/HaaS/blob/status-and-logs-fix/site.py#L187-L232) will blindly attempt to start a container even if there are no files present.

File: site.py [hunk 5]
```diff
@@ -315,11 +320,19 @@ def check_container(cid):
...
+        return render_template('status.html', notfound=True)
+
+    url = launch_container(uuid)
+    cid = container_id(uuid)[:16]
+    url = '/check_container/' + cid
     stdout = docker.logs(container=cid, stdout=True, stderr=False)
     stderr = docker.logs(container=cid, stderr=True, stdout=False)
-    return render_template('status.html', cid=cid, url=url, out=stdout, err=stderr)
+
+   return render_template('status.html', uuid=uuid, cid=cid,
```
Comment (bruth) [inline site.py]: Fix indent.

**Pull request merged**